PukingPredator/Assets/Scripts/UI/Note.cs

```diff
@@ -0,0 +1,125 @@
+using System.Collections;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Note : MonoBehaviour
+{
+    /// <summary>
+    /// The rate that the image fades in and out.
+    /// </summary>
+    [SerializeField]
+    private float fadeSpeed = 4f;
+
+    /// <summary>
+    /// If the player was nearby the last time a check was performed.
+    /// </summary>
+    private bool isPlayerNearby = false;
+
+    /// <summary>
+    /// To store the original alpha of the image.
+    /// </summary>
+    private float originalAlpha;
+
+    /// <summary>
+    /// The image shown by the note.
+    /// </summary>
+    [SerializeField]
+    private Sprite sprite;
+
+    /// <summary>
+    /// A reference to the image this note is for.
+    /// </summary>
+    private Image UIImage;
+
+
+
+    void Start()
+    {
+        CreateImageComponent();
+    }
+
+    void OnTriggerEnter(Collider other)
+    {
+        if (isPlayerNearby) { return; }
+
+        if (other.gameObject.CompareTag(GameTag.player))
+        {
+            isPlayerNearby = true;
+            StopAllCoroutines(); // Stop any ongoing fading out
+            StartCoroutine(FadeImage(originalAlpha)); // Fade in to alpha = 1 * originalAlpha
+        }
+    }
+
+    void OnTriggerExit(Collider other)
+    {
+        if (!isPlayerNearby) { return; }
+
+        if (other.gameObject.CompareTag(GameTag.player))
+        {
+            isPlayerNearby = false;
+            StopAllCoroutines(); // Stop any ongoing fading in
+            StartCoroutine(FadeImage(0f)); // Fade out to alpha = 0
+        }
+    }
+
+
+
+    /// <summary>
+    /// Create the Image component dynamically.
+    /// </summary>
+    void CreateImageComponent()
+    {
+        var canvas = GameObject.Find("Canvas");
+
+        // Create a new GameObject and make it a child of the Canvas
+        GameObject imageObject = new GameObject("ProximityImage");
```

Comment:
if we have multiple notes, this would result in multiple "ProximityImage" gameobjects. Is that fine?

Reply:
Its not an issue, they each have their own reference to the exact image they are linked to, and its not that important to manually look at those objects via the unity inspector